src/middlewared/middlewared/plugins/container_runtime_interface/containerd_utils.py

```diff
@@ -0,0 +1,11 @@
+import contextlib
+
+from .client import ContainerdClient
+
+
+def check_containerd_connection() -> bool:
+    with contextlib.suppress(Exception):
+        with ContainerdClient('image') as client:
+            client.get_image('alpine')
```

Comment:
Is there a easier/faster operation to check containerd health? Or is this image always present even on just-initialized system?

Reply:
We can use maybe https://hub.docker.com/r/rancher/pause 
This image is used by k3s anyway and its 290kb (compared to alpine ~3mb)

Reply:
If the image is not present, it returns nothing - so it is faster that way in terms of performance rather then us listing containers etc. I think we should be good to go as is

Reply:
Ah checks for presence. That should be fine yea!

Reply:
this seems flawed. what happens when the network isn't up but containerd is functional? You're trading 1 problem for another.

Reply:
Which network ?

Reply:
The truenas network? If this check gets called before the truenas network is setup, then this will return False because it can't get out to the internet.

Reply:
How this works is the following:

1. This request is definitely made after system is in ready state
2. This just connects to containerd socket and there is no outbound network call involved here - if the image exists, great - if not even then great as it is not pulling or anything, just checking it's existence on the filesystem

To conclude, that concern does not apply

Reply:
what a strange name...`client.get_image` managed to confuse 3 different developers....are we sure this doesn't go out to the network at all?